<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="KLRhodes" />

<meta name="date" content="2019-02-17" />

<title>MarkingDuplicateFragments</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<div class="container-fluid main-container">

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CM_ATAC_Workflow</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/KLRhodes/CM_ATAC_Workflow">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">MarkingDuplicateFragments</h1>
<h4 class="author"><em>KLRhodes</em></h4>
<h4 class="date"><em>2019-02-17</em></h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#report"> Report <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-02-18
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>CM_ATAC_Workflow/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.2.0). The <em>Report</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="report" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190216code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190216)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190216code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190216)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomKLRhodesCMATACWorkflowtree573d9cb141ef3f2eb936217da3cd09ec4bd8a959targetblank573d9cba"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/KLRhodes/CM_ATAC_Workflow/tree/573d9cb141ef3f2eb936217da3cd09ec4bd8a959" target="_blank">573d9cb</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomKLRhodesCMATACWorkflowtree573d9cb141ef3f2eb936217da3cd09ec4bd8a959targetblank573d9cba" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    docs/figure/

Untracked files:
    Untracked:  code/CollateFlagstat.R
    Untracked:  code/MarkDupFlagstatNoLoops.bash
    Untracked:  code/submit-MarkDupFlagstat.sh
    Untracked:  output/ATACQCsamplenamelineday.txt
    Untracked:  output/flagstat_all.txt

Unstaged changes:
    Modified:   _workflowr.yml

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KLRhodes/CM_ATAC_Workflow/blob/573d9cb141ef3f2eb936217da3cd09ec4bd8a959/analysis/MarkingDuplicateFragments.Rmd" target="_blank">573d9cb</a>
</td>
<td>
KLRhodes
</td>
<td>
2019-02-18
</td>
<td>
wflow_publish(“analysis/MarkingDuplicateFragments.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KLRhodes/CM_ATAC_Workflow/8d42fdd970a181d42bc6a7926b59f9cf3cab532c/docs/MarkingDuplicateFragments.html" target="_blank">8d42fdd</a>
</td>
<td>
KLRhodes
</td>
<td>
2019-02-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KLRhodes/CM_ATAC_Workflow/blob/1c7115d364946ff84b3d3985e5e7fa10ad7073a9/analysis/MarkingDuplicateFragments.Rmd" target="_blank">1c7115d</a>
</td>
<td>
KLRhodes
</td>
<td>
2019-02-17
</td>
<td>
wflow_publish(“analysis/MarkingDuplicateFragments.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KLRhodes/CM_ATAC_Workflow/1de8b535466368d9096b4e304c947912d37b9016/docs/MarkingDuplicateFragments.html" target="_blank">1de8b53</a>
</td>
<td>
KLRhodes
</td>
<td>
2019-02-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KLRhodes/CM_ATAC_Workflow/blob/3d4fb6fd7fc967ef19b798c05dbbc2f539c7b2bf/analysis/MarkingDuplicateFragments.Rmd" target="_blank">3d4fb6f</a>
</td>
<td>
KLRhodes
</td>
<td>
2019-02-17
</td>
<td>
wflow_publish(“analysis/MarkingDuplicateFragments.Rmd”)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="marking-duplicates-in-bam-files" class="section level2">
<h2>Marking Duplicates in bam files</h2>
<div id="the-bam-files-that-i-aligned-with-subread-do-not-have-duplicates-marked.-these-steps-mark-duplicates-and-eventually-output-text-files-containing-all-of-the-stats-that-come-out-of-samtools-flagstat" class="section level4">
<h4>The bam files that I aligned with subread do not have duplicates marked. These steps mark duplicates and eventually output text files containing all of the stats that come out of samtools flagstat</h4>
<p>Ran on test Midway2, Gilad partition with 12G of memory 2/17/19 using only one bam file</p>
<p>takes about 25 min per bam file</p>
<p>MarkDupFlagstatNoLoops.sh can be found in code directory– (will find a way to echo it here)</p>
<pre class="r"><code>#show code from MarkDupFlagstatNoLoops.sh
noquote(readLines(&quot;../code/MarkDupFlagstatNoLoops.bash&quot;))</code></pre>
<pre><code> [1] #!/bin/bash                                                                                                       
 [2]                                                                                                                   
 [3] f=$1                                                                                                              
 [4]                                                                                                                   
 [5] echo $f                                                                                                           
 [6]                                                                                                                   
 [7]                                                                                                                   
 [8] #have bam file from subread align                                                                                 
 [9] #need to name sort the bam file                                                                                   
[10] echo &quot;name sorting&quot;                                                                                               
[11] samtools sort -n $f -o ./FlagStat/${f/.bam/.namesorted.bam}                                                       
[12]                                                                                                                   
[13] #need to use fixmate to add matescore tags                                                                        
[14] echo &quot;fixmate&quot;                                                                                                    
[15] samtools fixmate -m ./FlagStat/${f/.bam/.namesorted.bam} ./FlagStat/${f/.bam/.fixed.namesorted.bam}               
[16]                                                                                                                   
[17] #need to get position sorted file to feed into markdup                                                            
[18] echo &quot;position sorting&quot;                                                                                           
[19] samtools sort -o ./FlagStat/${f/.bam/.fixed.positionsorted.bam} ./FlagStat/${f/.bam/.fixed.namesorted.bam}        
[20]                                                                                                                   
[21] #now use markdup to mark duplicate alignments. -l 600 indicates the expected maximum read length. I dont          
[22] #know if this means read length or insert length or fragment length for PE data. -s prints short summary. can add 
[23] #-r to actually REMOVE the duplicates                                                                             
[24] echo &quot;marking duplicates&quot;                                                                                         
[25] samtools markdup -l 600 -s ./FlagStat/${f/.bam/.fixed.positionsorted.bam} ./FlagStat/${f/.bam/.markdup.bam}       
[26]                                                                                                                   
[27] #run samtools flagstat to get text file output that includes duplicate count                                      
[28] echo &quot;running flagstat&quot;                                                                                           
[29] samtools flagstat ./FlagStat/${f/.bam/.markdup.bam} &gt; ./FlagStat/${f/.bam/.flagstat.txt}                          
[30]                                                                                                                   </code></pre>
<pre class="r"><code>#from the CM_TC_ATAC/19209_and_19238/bam directory
sbatch --partition=gilad --mem=4G submit-MarkDupFlagstat.sh</code></pre>
<p>Now I have separate txt files with the flagstat output for each sample. I want to pull out total fragments, mapped fragments, and duplicate count.</p>
<p>Did this using CollateFlagstat.R</p>
<pre class="r"><code>noquote(readLines(&quot;../code/CollateFlagstat.R&quot;))</code></pre>
<pre><code>Warning in readLines(&quot;../code/CollateFlagstat.R&quot;): incomplete final line
found on &#39;../code/CollateFlagstat.R&#39;</code></pre>
<pre><code> [1] #!/usr/bin/env Rscript                                                                                
 [2]                                                                                                       
 [3] #Collate flagstat metrics from each flagstat.txt file by running this script on all and appending info
 [4] #to a new txt                                                                                         
 [5]                                                                                                       
 [6]                                                                                                       
 [7] # Obtain name of fastq file passed at the command line                                                
 [8] args &lt;- commandArgs(trailingOnly = TRUE)                                                              
 [9] stopifnot(length(args) == 1)                                                                          
[10] flag &lt;- args[1]                                                                                       
[11] stopifnot(file.exists(flag))                                                                          
[12]                                                                                                       
[13] #read in the text file, separate at plus sign for ease                                                
[14] txt&lt;- read.csv( flag, header=F, sep=&quot;+&quot;)                                                              
[15]                                                                                                       
[16] value&lt;- txt$V1                                                                                        
[17] flagstat&lt;- gsub(&quot;0 &quot;, &quot;&quot;,txt$V2)                                                                      
[18] file&lt;- basename(flag)                                                                                 
[19]                                                                                                       
[20] info=noquote(cbind(file, flagstat, value))                                                            
[21] colnames(info)&lt;- c(&quot;sample&quot;, &quot;flagstat_metric&quot;, &quot;value&quot;)                                              
[22]                                                                                                       
[23] write.table(info, file=&quot;flagstat_all.txt&quot;, append=TRUE, sep=&quot;,&quot;, quote=FALSE,                         
[24] \trow.names=FALSE, col.names = !file.exists(&quot;flagstat_metrics.txt&quot;))                                   
[25] \t                                                                                                     
[26] \t                                                                                                     </code></pre>
<pre class="r"><code>#Loops through all .flagstat.txt files in the directory
sbatch --partition=gilad --mem=4G submit-CollateFlagstat.sh</code></pre>
<p>Copied flagstat_all.txt to local for analysis. It exists in the ouput folder of the workflowr directory</p>
<pre class="r"><code>all&lt;- read.csv(&quot;../output/flagstat_all.txt&quot;, header=T, sep=&quot;,&quot;, stringsAsFactors = F)
#sampleinf&lt;- read.table(&quot;../output/ATACQCsamplenamelineday.txt&quot;, header=T)


all$value&lt;- as.numeric(as.character(all$value))</code></pre>
<pre><code>Warning: NAs introduced by coercion</code></pre>
<pre class="r"><code>#all&lt;- all[order(all$sample),]

dupcount&lt;- all[all$flagstat_metric ==&quot; duplicates&quot;,]
mappedcount&lt;- all[grep(&quot; mapped \\(&quot;, all$flagstat_metric),]

#plot the number of mapped fragments
barplot(mappedcount$value)+
  abline(h=20000000, col=&quot;red&quot;)</code></pre>
<p><img src="figure/MarkingDuplicateFragments.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>numeric(0)</code></pre>
<pre class="r"><code>#get nonredundant fraction (number of unique fragments/total fragments)
nrf&lt;- (mappedcount$value-dupcount$value)/mappedcount$value
hist(nrf)</code></pre>
<p><img src="figure/MarkingDuplicateFragments.Rmd/unnamed-chunk-5-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#is nrf different between the two lines?
barplot(nrf)+
  abline(h=mean(nrf), col=&quot;blue&quot;)</code></pre>
<p><img src="figure/MarkingDuplicateFragments.Rmd/unnamed-chunk-5-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>numeric(0)</code></pre>
<pre class="r"><code>#most are still very low. the undetermined samples have high non redundant fraction, but this is not surprising because they&#39;ll have reads from all the samples in the lane.Notably, Nick also had lots of duplicate reads 
#(see Figure S5 from genome biology paper--need to check on the final usable fragment count though)</code></pre>
<div class="figure">
<img src="assets/BanoFigS5.jpg" alt="Banovich 2018 Figure S5" />
<p class="caption">Banovich 2018 Figure S5</p>
</div>
<pre class="r"><code>#get the number of unique mapped fragments per sample
uniquemappedfrags&lt;- (mappedcount$value-dupcount$value)
barplot(uniquemappedfrags) +
abline(h=10000000, col=&quot;red&quot;)</code></pre>
<p><img src="figure/MarkingDuplicateFragments.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>numeric(0)</code></pre>
<p>Next steps: remove duplicates, remove MT reads Should I run fragment size distribution on the cleaned bams or on original bams?</p>
<br> <br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.2 (2018-12-20)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 17134)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252 
[2] LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] workflowr_1.2.0 Rcpp_1.0.0      digest_0.6.18   rprojroot_1.3-2
 [5] backports_1.1.3 git2r_0.24.0    magrittr_1.5    evaluate_0.12  
 [9] stringi_1.2.4   fs_1.2.6        whisker_0.3-2   rmarkdown_1.11 
[13] tools_3.5.2     stringr_1.3.1   glue_1.3.0      xfun_0.4       
[17] yaml_2.2.0      compiler_3.5.2  htmltools_0.3.6 knitr_1.21     </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
