---
title: "MarkingDuplicateFragments"
author: "KLRhodes"
date: "2019-02-17"
output: workflowr::wflow_html
---
##Marking Duplicates in bam files

####The bam files that I aligned with subread do not have duplicates marked. These steps mark duplicates and eventually output text files containing all of the stats that come out of samtools flagstat


Ran on  test Midway2, Gilad partition with 12G of memory 2/17/19 using only one bam file

takes about 25 min per bam file

MarkDupFlagstatNoLoops.sh can be found in code directory-- (will find a way to echo it here)
```{r}
#show code from MarkDupFlagstatNoLoops.sh
noquote(readLines("../code/MarkDupFlagstatNoLoops.bash"))
```

```{r eval=F}
#from the CM_TC_ATAC/19209_and_19238/bam directory
sbatch --partition=gilad --mem=4G submit-MarkDupFlagstat.sh
```

Now I have separate txt files with the flagstat output for each sample. I want to pull out total READs, mapped READs, and duplicate count.  

Did this using CollateFlagstat.R
```{r}
noquote(readLines("../code/CollateFlagstat.R"))
```

```{r eval=F}
#Loops through all .flagstat.txt files in the directory
sbatch --partition=gilad --mem=4G submit-CollateFlagstat.sh
```

####Flagstat Content Guide:

#####in total (QC-passed reads + QC-failed reads)
    -total READS before mapping
#####secondary
#####supplementary
#####duplicates
    -duplicate count (as marked by samtools markdup)
#####mapped
    -total mapped READS
#####paired in sequencing
  -how many reads had a pair. This is all of them because aligning Read 1 and Read 2 necessitates that they be "paired"
#####read1
  -how many "read 1" READs mapped
#####read2
  -how many "read 2" READs mapped
  (should be the same as read1)
#####properly paired 
  -how many R1 R2 are "properly paired" based on orientation, distance
#####with itself and mate mapped
#####singletons 
#####with mate mapped to a different chr
#####with mate mapped to a different chr (mapQ>=5)




Copied flagstat_all.txt to local for analysis. It exists in the ouput folder of the workflowr directory
```{r}
library(ggplot2)
library(RColorBrewer)
```

```{r }
all<- read.csv("../output/flagstat_all.txt", header=T, sep=",", stringsAsFactors = F)
#sampleinf<- read.table("../output/ATACQCsamplenamelineday.txt", header=T)


all$value<- as.numeric(as.character(all$value))

#all<- all[order(all$sample),]

dupcount<- all[all$flagstat_metric ==" duplicates",]
mappedcount<- all[grep(" mapped \\(", all$flagstat_metric),]
totalcount<-all[grep("total", all$flagstat_metric),]

#plot the number of mapped reads
barplot(mappedcount$value)+
  abline(h=20000000, col="red")

#get nonredundant fraction (number of unique reads/total reads)
nrf<- (mappedcount$value-dupcount$value)/mappedcount$value
hist(nrf)
#is nrf different between the two lines?
barplot(nrf)+
  abline(h=mean(nrf), col="blue")
#most are still very low. the undetermined samples have high non redundant fraction, but this is not surprising because they'll have reads from all the samples in the lane.

#get the number of unique mapped reads per sample
uniquemappedreads<- (mappedcount$value-dupcount$value)
barplot(uniquemappedreads) +
abline(h=10000000, col="red")

#make a data frame containing ABSOLUTE mapped reads and  ABSOLUTE duplicated reads
mapped<- rep("mapped", 34)
uniq<- rep("unique", 34)
tot<- rep("total", 34)

mapdf<- cbind(as.character(mappedcount$sample), mapped, mappedcount$value)

unidf<- cbind(as.character(mappedcount$sample), uniq, mappedcount$value-dupcount$value)

totdf<- cbind(as.character(mappedcount$sample), tot, totalcount$value)


comb<-as.data.frame(rbind(totdf, mapdf, unidf))
colnames(comb)<-c("Sample", "Metric", "Value")


p<- (ggplot(comb,aes(x=as.character(comb$Sample), y=as.numeric(as.character(comb$Value)), fill=as.character(comb$Metric)))+ 
        geom_bar(stat="identity",position = "identity", alpha=.5))+
        ylim(0, 65000000)+
        ylab("Absolute read Count")+
        xlab("Sample (not ordered)")+
        ggtitle("Absolute reads, Mapped read, and Unique read", subtitle = "(reads not marked as duplicated by markdup)") +
        geom_hline(yintercept = 10000000)

p+scale_fill_brewer(breaks=c("total", "mapped", "unique"), type="qual", palette="Accent")


#plot fractions
comb$Value<-(as.numeric(as.character(comb$Value))/(rep(totalcount$value, 3)))

f<- (ggplot(comb,aes(x=as.character(comb$Sample), y=as.numeric(as.character(comb$Value)), fill=as.character(comb$Metric)))+ 
        geom_bar(stat="identity",position = "identity", alpha=.5))+
        ylim(0, 1)+
        ylab("Absolute read Count")+
        xlab("Sample (not ordered)")+
        ggtitle("Percent Mapped reads and Unique reads", subtitle = "(reads not marked as duplicated by markdup)")
f+scale_fill_brewer(breaks=c("total", "mapped", "unique"), type="qual", palette="Accent")



#Nick also had lots of duplicate reads
#(see Figure S5 from genome biology paper)
```
![Banovich 2018 Figure S5](assets/BanoFigS5.jpg)




